src/components/PeoplePage.tsx

```diff
@@ -0,0 +1,125 @@
+import React, { useCallback, useEffect, useState } from 'react';
+import { Loader } from './Loader/Loader';
+import { getPeople } from '../api';
+import { Person } from '../types';
+import { PersonLink } from './PersonLink';
+import classNames from 'classnames';
+import { useParams } from 'react-router-dom';
+
+export const PeoplePage = () => {
+  const { slug } = useParams();
+
+  const [people, setPeople] = useState<Person[]>([]);
+  const [error, setError] = useState<string | null>(null);
+  const [loading, setLoading] = useState(true);
+  const [selectedPerson, setSelectedPerson] = useState<string | null>(
+    slug || null,
+  );
+
+  useEffect(() => {
+    const fetchPeople = async () => {
+      try {
+        const data = await getPeople();
+
+        setPeople(data);
+      } catch {
+        setError('Something went wrong');
+      } finally {
+        setLoading(false);
+      }
+    };
+
+    fetchPeople();
+  }, []);
+
+  const findPersonByName = useCallback(
+    (name: string | undefined | null): Person | null =>
+      name
+        ? people.find(p => p.name.toLowerCase() === name.toLowerCase()) || null
+        : null,
+    [people],
+  );
```

Comment:
The `findPersonByName` function is defined using `useCallback`, which is good for performance optimization. However, it might be unnecessary unless you are passing this function to a child component that relies on referential equality. If it's only used within this component, you can define it as a regular function without `useCallback`.